### PR TITLE
Add averagepool test case

### PR DESCRIPTION
**Pull request by fumihwh**

### CODE REVIEW

**Comment by houseroad** [comment]
@onnxbot test this please

**Comment by houseroad** [comment]
@onnxbot add to whitelist

**Comment by houseroad** [comment]
@onnxbot test this please

File: docs/Changelog.md [hunk 1]
```diff
@@ -2960,7 +2969,16 @@ opset_import {
    the tensor according to kernel sizes, stride sizes, and pad lengths.
    max pooling consisting of computing the max on all values of a
    subset of the input tensor according to the kernel size and downsampling the
-   data into the output tensor Y for further processing.
+   data into the output tensor Y for further processing. The output spatial shape will be following:
+   ```
+   VALID or pads given: output_spatial_shape[i] = ceil((input_spatial_shape[i] - (kernel_spatial_shape[i] - 1)) / strides_spatial_shape[i]))
+   SAME_UPPER or SAME_LOWER: output_spatial_shape[i] = ceil(input_spatial_shape[i] / strides_spatial_shape[i]))
```
Comment (houseroad) [inline docs/Changelog.md]: Please notice that: auto_pad is a deprecated attribute, we encourage users to use explicit padding.

Please also add how we calculate the output_shape, when we use explicit padding:
Some reference is here:
http://pytorch.org/docs/master/nn.html#torch.nn.AvgPool2d
I think most frameworks compute the output shape using this formula. (Checked Caffe2, PyTorch, MXNet)

File: onnx/backend/test/case/node/averagepool.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    output_shape: [1, 1, 2, 2]
+    """
+    node = onnx.helper.make_node(
+      'AveragePool',
+      inputs=['x'],
+      outputs=['y'],
+      kernel_shape=[2, 2],
+      strides=[2, 2]
+    )
+    x = np.array([[[
```
Comment (houseroad) [inline onnx/backend/test/case/node/averagepool.py]: One comment here: input_shape (1, 1, 5, 5)

File: onnx/backend/test/case/node/averagepool.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    output_shape: [1, 1, 3, 3]
+    pad_shape: [2, 2] -> [1, 1, 1, 1] by axis
+    """
+    node = onnx.helper.make_node(
+      'AveragePool',
+      inputs=['x'],
+      outputs=['y'],
+      kernel_shape=[3, 3],
+      strides=[2, 2],
+      auto_pad='SAME_UPPER'
```
Comment (houseroad) [inline onnx/backend/test/case/node/averagepool.py]: Because auto_pad is deprecated, please provide (additional) examples with explicit padding.

File: docs/Changelog.md (hunk 1 above)
Comment (fumihwh) [reply docs/Changelog.md]: @houseroad Actually, two formulas are same. ( I just ignored I should add pad_spatial to input_spatial_shape.) Anyway, I will fix it.

**Comment by houseroad** [comment]
@onnxbot retest this please

**Comment by houseroad** [comment]
@fumihwh sorry for the late review. Overall, the code looks good. Only one thing you probably should add to description of the AveragePool op: when computing the average of one window, the pad is not counted into the divisor. (I think most of the existing models implement this way, also your code does the same thing.)

**Comment by fumihwh** [comment]
@houseroad AveragePool and MaxPool are using same base doc.
If I want to add description for AveragePool, I have two choices:
1. Copy and paste to make individual AveragePool and MaxPool ONNX_OPERATOR_SCHEMA
2. Add arg to PoolOpSchemaGenerator called `additional description`

What do you think?

**Comment by houseroad** [comment]
@fumihwh sure, feel free to do it, as long as it is easy to use and maintain.
